lib/PuppeteerSharp/Browser.cs

```diff
@@ -352,98 +441,44 @@ private async void Connection_Disconnected(object sender, EventArgs e)
             }
         }
 
-        private async void Connect_MessageReceived(object sender, MessageEventArgs e)
+        private Task AttachAsync()
         {
-            try
-            {
-                switch (e.MessageID)
-                {
-                    case "Target.targetCreated":
-                        await CreateTargetAsync(e.MessageData.ToObject<TargetCreatedResponse>(true)).ConfigureAwait(false);
-                        return;
-
-                    case "Target.targetDestroyed":
-                        await DestroyTargetAsync(e.MessageData.ToObject<TargetDestroyedResponse>(true)).ConfigureAwait(false);
-                        return;
-
-                    case "Target.targetInfoChanged":
-                        ChangeTargetInfo(e.MessageData.ToObject<TargetCreatedResponse>(true));
-                        return;
-                }
-            }
-            catch (Exception ex)
-            {
-                var message = $"Browser failed to process {e.MessageID}. {ex.Message}. {ex.StackTrace}";
-                _logger.LogError(ex, message);
-                Connection.Close(message);
-            }
+            Connection.Disconnected += Connection_Disconnected;
+            TargetManager.TargetAvailable += TargetManager_TargetAvailable;
+            TargetManager.TargetGone += TargetManager_TargetGone;
+            TargetManager.TargetChanged += TargetManager_TargetChanged;
+            TargetManager.TargetDiscovered += TargetManager_TargetDiscovered;
+            return TargetManager.InitializeAsync();
         }
 
-        private void ChangeTargetInfo(TargetCreatedResponse e)
+        private void Detach()
```

Comment:
I noticed that `Detach` is not referenced anywhere.
Where should it be invoked from?

Reply:
True. I ask for confirmation [upstream](https://github.com/puppeteer/puppeteer/pull/9782)